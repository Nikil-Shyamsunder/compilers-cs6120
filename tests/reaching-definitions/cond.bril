@main {
  a: int = const 47;
  b: int = const 42;
  c: int = const 66; # value is always overridden
  cond: bool = const true;
  br cond .left .right;
.left:
  b: int = const 1;
  c: int = const 5;
  jmp .end;
.right:
  a: int = const 2;
  c: int = const 10;
  jmp .end;
.end:
  d: int = sub a c;
  print d;
}