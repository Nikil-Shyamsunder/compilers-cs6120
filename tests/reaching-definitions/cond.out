Block right:
  b defined at {"dest":"b","op":"const","type":"int","value":42}
  cond defined at {"dest":"cond","op":"const","type":"bool","value":true}
  a defined at {"dest":"a","op":"const","type":"int","value":2}
  c defined at {"dest":"c","op":"const","type":"int","value":10}
Block end:
  c defined at {"dest":"c","op":"const","type":"int","value":10}
  a defined at {"dest":"a","op":"const","type":"int","value":2}
  b defined at {"dest":"b","op":"const","type":"int","value":42}
  a defined at {"dest":"a","op":"const","type":"int","value":47}
  b defined at {"dest":"b","op":"const","type":"int","value":1}
  cond defined at {"dest":"cond","op":"const","type":"bool","value":true}
  d defined at {"args":["a","c"],"dest":"d","op":"sub","type":"int"}
  c defined at {"dest":"c","op":"const","type":"int","value":5}
Block left:
  c defined at {"dest":"c","op":"const","type":"int","value":5}
  cond defined at {"dest":"cond","op":"const","type":"bool","value":true}
  b defined at {"dest":"b","op":"const","type":"int","value":1}
  a defined at {"dest":"a","op":"const","type":"int","value":47}
Block main-block0:
  a defined at {"dest":"a","op":"const","type":"int","value":47}
  c defined at {"dest":"c","op":"const","type":"int","value":66}
  b defined at {"dest":"b","op":"const","type":"int","value":42}
  cond defined at {"dest":"cond","op":"const","type":"bool","value":true}
